src/meta/src/stream/source_manager.rs

```diff
@@ -93,12 +85,29 @@ where
     pub async fn schedule_split_for_actors(
         &self,
         actors: HashMap<SourceId, Vec<Vec<ActorId>>>,
+        affiliated_source: Option<Source>,
     ) -> Result<HashMap<ActorId, Vec<SplitImpl>>> {
-        let source_splits = try_join_all(
-            actors
-                .keys()
-                .map(|source_id| self.fetch_splits_for_source(*source_id)),
-        )
+        let source_ref = &affiliated_source;
+        let source_splits = try_join_all(actors.keys().map(|source_id| async move {
+            let catalog_guard = self.catalog_manager.get_catalog_core_guard().await;
```

Comment:
Estimate affiliated source firstly could cover all cases of materialized source. This operation can avoid locking catalog.

Reply:
Refactored.